libs/wingc/src/type_check.rs

```diff
@@ -1460,15 +1457,17 @@ impl<'a> TypeChecker<'a> {
 
 				// Verify that no unexpected fields are present
 				for (name, _t) in field_types.iter() {
-					if !st.env.lookup(&name, Some(statement_idx)).is_ok() {
+					if !st.env.lookup(&name, Some(self.statement_idx)).is_ok() {
 						self.expr_error(exp, format!("\"{}\" is not a field of \"{}\"", name.name, st.name.name));
 					}
 				}
 
 				struct_type
 			}
 			ExprKind::JsonLiteral { is_mut, element } => {
-				self.type_check_exp(&element, env, statement_idx, &TypeCheckerContext { in_json: true });
+				self.in_json = true;
+				self.type_check_exp(&element, env);
+				self.in_json = false;
```

Comment:
Almost, if I do a json literal within a json literal I think I'll need a counter
```rust
self.in_json += 1;
self.type_check_expt();
self.i_json -=1;
```
I didn't do this because I was under the impression we don't allow an explicit `Json` within a `Json`. But I guess we do. So I'll fix.

Reply:
I think if you use a boolean it ought to still work since we are using the call stack to track the history of `in_json` values, but either should be OK

Reply:
once `in_json` is in (mut) `self` we're not using the call stack anymore.